research-software-directory/RSD-as-a-service
# Initial frontend setup

# Initial setup frontend with NextJS, MUI-5, Tailwind and Jest

Adresses #42 #43 #44 

How to test:
- in the **frontend folder** copy `env.local.example` to `env.local`. The content of this file is send to you via Teams or email. If not, ask one of core developers to provide you content of env.local file.
- `docker-compose up --build` in the root of the project should start all parts of RSD. On `http://localhost:3000` you should see home page (image below). To stop use `Ctrl +C`
-  you can navigate to: software and project pages (these have only page title)
-  you can try to login with SURFconext `test environment`:  when redirected to SURF login page select `SURFconext Test Idp` provider and specify user `professor1` and pass `professor1`. After authentication you will be send back to user/profile page by default.  
- unit testing with Jest: see readme file in the frontend directory for more info. **You need Node.js and Yarn installed locally on your machine**. When running docker-compose the tests are runned in the container before the build.

## Expected result after docker-compose up
![image](https://user-images.githubusercontent.com/9204081/145427552-8285bc71-e201-415e-b037-3c2e139a4b8b.png)
